File: src/main/java/com/example/demo/controller/UserController.java
package com.example.demo.controller;

import com.example.demo.auth.Auth;
import com.example.demo.model.User;
import com.example.demo.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/users")
public class UserController {
  private UserService userService;

  @Autowired
  public UserController(UserService userService) {
    this.userService = userService;
  }

  @GetMapping
  public Iterable<User> getUsers() {
    return userService.getUsers();
  }

  @PostMapping
  public User createUser(@RequestBody User user) {
    return userService.createUser(user);
  }

  @Auth
  @GetMapping("/me")
  public User getCurrentUser() {
    return userService.getCurrentUser();
  }
}

File: src/main/kotlin/com/example/demo/controller/UserController.kt
package com.example.demo.controller

import com.example.demo.auth.Auth
import com.example.demo.model.User
import com.example.demo.service.UserService
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

@RestController
@RequestMapping("/api/users")
class UserController(private val userService: UserService) {
  @get:GetMapping
  val users: Iterable<User>
    get() = userService.users

  @PostMapping
  fun createUser(@RequestBody @Valid user: User): User {
    return userService.createUser(user)
  }

  @get:GetMapping("/me")
  @get:Auth
  val currentUser: User?
    get() = userService.currentUser
}
